### PR TITLE
AWS updates

### PR DESCRIPTION
**Pull request by rvelaVenafi**
Added support for TPP Token Authentication.
Migrated the Cloud API to OutagePredict.

### CODE REVIEW

File: request/main_test.go [hunk 1]
```diff
@@ -13,30 +13,36 @@ import (
...
 )
 
 const (
 	wrongResponseCode = "Request returned code: %d message: %s"
-)
-const (
-	acmpcaIssueCertificateRequest = `{
+
+	acmpcaListCertificateAuthoritiesRequest = `{"MaxResults": 100}`
+	acmpcaIssueCertificateRequest           = `{
```
Comment (marcos-albornoz) [inline request/main_test.go:29]: can you remove the extra blank spaces?

File: policy/main.go [hunk 2]
```diff
@@ -97,48 +100,128 @@ func main() {
...
+			},
 		}
-	} else if cloudKey != "" {
+	} else if apiKey != "" {
 		config = vcert.Config{
 			ConnectorType: endpoint.ConnectorTypeCloud,
-			Credentials:   &endpoint.Authentication{APIKey: cloudKey},
-			BaseUrl:       cloudUrl,
+			Credentials: &endpoint.Authentication{
+				APIKey: apiKey,
```
Comment (angelmoo) [inline policy/main.go:159]: Not Sure if is github editor or you may need to format this part of the code,

File: request/main_test.go (hunk 1 above)
Comment (rvelaVenafi) [reply request/main_test.go:29]: That is a change made by the go formatter (go-fmt). Even if I remove the blank spaces, when trying to build Go will complain and request that I run go-fmt again and add the spaces back.

File: policy/main.go (hunk 2 above)
Comment (rvelaVenafi) [reply policy/main.go:159]: APIKey is a field of the Authentication struct, thats why it has more indentation.

**Review by angelmoo — APPROVED** [review]
Changes looks good.

File: policy/main.go [hunk 3]
```diff
@@ -97,48 +100,128 @@ func main() {
...
 		}
-		if trustBundle != "" {
-			buf, err := base64.StdEncoding.DecodeString(trustBundle)
-			if err != nil {
-				log.Printf("Can`t read trust bundle from file %s: %v\n", trustBundle, err)
-				return nil, err
-			}
-			config.ConnectionTrust = string(buf)
+
+	} else if tppUrl != "" && (accessToken != "" || refreshToken != "") {
```
Comment (tr1ck3r) [inline policy/main.go:145]: Should this condition come first?  Tokens should always take precedence over username/password for the case where a user (mistakenly) specifies both.

**Review by marcos-albornoz — APPROVED** [review]

File: policy/main_test.go [hunk 4]
```diff
@@ -23,19 +23,36 @@ func TestHandleRequestTPP(t *testing.T) {
...
 		t.Fatal(err)
 	}
 
-	vcertConnector, err = getConnection(os.Getenv("TPPURL"), os.Getenv("TPPUSER"), os.Getenv("TPPPASSWORD"), "", "", base64.StdEncoding.EncodeToString(trust_bundle))
+	vcertConnector, err = getConnection(os.Getenv("TPPURL"), os.Getenv("TPPUSER"), os.Getenv("TPPPASSWORD"), "", "", "", base64.StdEncoding.EncodeToString(trustBundle))
 	if err != nil {
 		t.Fatal(err)
 	}
-	testHandleRequest(t, os.Getenv("TPPZONE"), "UnexistedZone\\Olololololo", `^[\p{L}\p{N}-_*]+\.example\.com$`)
+	testHandleRequest(t, os.Getenv("TPPZONE"), "UnexistedZone\\Olololololo", ".*")
```
Comment (tr1ck3r) [inline policy/main_test.go:35]: "UnexistedZone" -> "InvalidZone" as you've done elsewhere

File: policy/main_test.go [hunk 5]
```diff
@@ -23,19 +23,36 @@ func TestHandleRequestTPP(t *testing.T) {
...
+	trustBundle, err := ioutil.ReadAll(f)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	vcertConnector, err = getConnection(os.Getenv("TPP_TOKEN_URL"), "", "", os.Getenv("TPP_ACCESS_TOKEN"), os.Getenv("TPP_REFRESH_TOKEN"), "", base64.StdEncoding.EncodeToString(trustBundle))
+	if err != nil {
+		t.Fatal(err)
+	}
+	testHandleRequest(t, os.Getenv("TPPZONE"), "UnexistedZone\\Olololololo", ".*")
```
Comment (tr1ck3r) [inline policy/main_test.go:52]: "UnexistedZone" -> "InvalidZone" as you've done elsewhere

**Review by tr1ck3r — CHANGES_REQUESTED** [review]
Looks good @rvelaVenafi just a couple of small things...

File: policy/main_test.go (hunk 4 above)
Comment (rvelaVenafi) [reply policy/main_test.go:35]: Changed

File: policy/main_test.go (hunk 5 above)
Comment (rvelaVenafi) [reply policy/main_test.go:52]: Changed

File: policy/main.go (hunk 3 above)
Comment (rvelaVenafi) [reply policy/main.go:145]: Token conditional moved up the hierarchy to take precedence over user/password

**Pull request merged**